gui/cppchecklibrarydata.cpp

```diff
@@ -33,12 +33,12 @@
 const unsigned int CppcheckLibraryData::Function::Arg::ANY = ~0U;
 const unsigned int CppcheckLibraryData::Function::Arg::VARIADIC = ~1U;
 
-static std::string unhandledElement(const QXmlStreamReader &xmlReader)
+NORETURN static void unhandledElement(const QXmlStreamReader &xmlReader)
 {
     throw std::runtime_error(QObject::tr("line %1: Unhandled element %2").arg(xmlReader.lineNumber()).arg(xmlReader.name().toString()).toStdString());
 }
 
-static std::string mandatoryAttibuteMissing(const QXmlStreamReader &xmlReader, const QString& attributeName)
+NORETURN static void mandatoryAttibuteMissing(const QXmlStreamReader &xmlReader, const QString& attributeName)
```

Comment:
The return values were never used and now also caused a confusing -Wunreachable-code warning in each invocation - see https://github.com/llvm/llvm-project/issues/152477.